base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -541,6 +546,12 @@ public void onCapabilityRetrieved(Object capability) {
                         int currentWindowID = windowCapability.getWindowID() != null ? windowCapability.getWindowID() : PredefinedWindows.DEFAULT_WINDOW.getValue();
                         if (currentWindowID == PredefinedWindows.DEFAULT_WINDOW.getValue()) {
                             defaultMainWindowCapability = windowCapability;
+
+                            KeyboardCapabilities keyboardCapabilities = windowCapability.getKeyboardCapabilities();
+                            if (keyboardCapabilities.getMaskInputCharactersSupported() != null
+                                    && !keyboardCapabilities.getMaskInputCharactersSupported()) {
+                                keyboardConfiguration.setMaskInputCharacters(KeyboardInputMask.DISABLE_INPUT_KEY_MASK);
+                            }
```

Comment:
```suggestion
```
Getting the `keyboardCapabilities` and checking it here is not necessary here and can be removed. We should check the capability when setting the keyboard configuration instead (this will be described in another comment)